autotest/pyscripts/test_gdal2tiles.py

```diff
@@ -101,10 +101,38 @@ def test_gdal2tiles_py_cleanup():
 
     return 'success'
 
+
+def test_does_not_error_when_source_bounds_close_to_tiles_bound():
+    """
+    Case where the geo coords of the image top Y point gives a corresponding tile of X, Y, Z
+    BUT where only a portion of the pixels of the first pixel line would be on this tile.
+    When computing the raster for the tile, we don't get any data (the pixel line is "assigned" to
+    the tile below)
+    """
+    out_folder = 'tmp/out_gdal2tiles_bounds_approx'
+    try:
+        shutil.rmtree(out_folder)
+    except:
+        pass
+
+    script_path = test_py_scripts.get_py_script('gdal2tiles')
+    if script_path is None:
+        return 'skip'
+
+    # Will blow an exception if the case is not handled properly
+    test_py_scripts.run_py_script(
```

Comment:
I don't think the code you quote is the one that will be run by test_py_scripts.run_py_script. As far as I can see in autotest/pymod/test_py_scripts.py, no exception is caught in run_py_script_as_py_module. Although it should probably caught exception to do the necessary cleanups and rethrow them.

One option if you are still stuck would be to use explicitly test_py_scripts.run_py_script_as_external_script() so as to launch a new Python process. Slower but no exception will every come from it !